mediorum/server/serve_blob.go

```diff
@@ -108,12 +108,27 @@ func (ss *MediorumServer) getBlobDoubleCheck(c echo.Context) error {
 	return c.JSON(200, existingBlob)
 }
 
+func (ss *MediorumServer) ensureNotDelisted(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		ctx := c.Request().Context()
+		key := c.Param("cid")
+
+		if ss.isCidBlacklisted(ctx, key) {
+			ss.logger.Info("cid is blacklisted", "cid", key)
+			return c.String(403, "cid is blacklisted by this node")
+		}
+
+		c.Set("checkedDelistStatus", true)
+		return next(c)
+	}
+}
+
 func (ss *MediorumServer) getBlob(c echo.Context) error {
 	ctx := c.Request().Context()
 	key := c.Param("cid")
 	logger := ss.logger.With("cid", key)
 
-	if ss.isCidBlacklisted(ctx, key) {
+	if !c.Get("checkedDelistStatus").(bool) && ss.isCidBlacklisted(ctx, key) {
```

Comment:
this will panic if the cast fails... like if it's not set in the context.

safer version is like:

```
if shouldCheck, exists := c.Get("checkedDelistStatus").(bool); exists && shouldCheck && ss.isCidBlacklisted(ctx, key)  {

}
```

but wondering why use this context variable if the middleware will always set it to true?  Is there ever a codepath where this is false?